citations-tool/tool/src/java/org/sakaiproject/citation/tool/CitationHelperAction.java

```diff
@@ -1083,9 +1084,10 @@ protected void doPostJsonResponse(ParameterParser params,
 			String addSectionHTML = params.getString("addSectionHTML");
 			int locationId = params.getInt("locationId");
 			String cleanAddSectionHTML = getFormattedText().processFormattedText(addSectionHTML, new StringBuilder(), true, true);
+			cleanAddSectionHTML = cleanAddSectionHTML.replaceAll("'", "&#8217;").replaceAll("\"", "&#8220;");
```

Comment:
The reason I'm replacing single and double quotes with their HTML entities is that they are used in the attributes of <li> tags and quotes screw up the HTML.  Escaping them by usual means (e.g., \' or \") does not work in HTML:  

http://stackoverflow.com/questions/2428572/how-to-escape-single-quote

I'll put a comment in the code.